### PR TITLE
feat: container-type level version compatibility check

### PR DESCRIPTION
**Pull request by KeranYang**

### CODE REVIEW

**Review by vigith — APPROVED** [review]

File: pynumaflow/sourcetransformer/server.py [hunk 1]
```diff
@@ -112,6 +113,10 @@ def start(self):
             self.sock_path,
             self.max_threads,
         )
+        serv_info = ServerInfo.get_default_server_info()
+        serv_info.minimum_numaflow_version = MINIMUM_NUMAFLOW_VERSION[
+            ContainerType.Sourcetransformer
+        ]
```
[the remark this answers is not in the document]
Comment (KeranYang) [reply pynumaflow/sourcetransformer/server.py:119]: yes, I am also curious about it. Some of the servers actually got covered, e.g., sink server. @kohlisid do you know why?

**Comment by kohlisid** [comment]
@KeranYang I see this got merged but to answer your question, in our code the server start utility would add the default server info automatically, 
```
    if server_info is None:
        # Add the server information to the server info file
        # if not provided
        server_info = ServerInfo.get_default_server_info()
```
Only if a deviation from default was required we could pass a custom info object

cc @vigith

**Comment by vigith** [comment]
I am thinking whether we use the same pattern we used in Rust SDK.

**Comment by kohlisid** [comment]
This is the similar patten I had added in Rust SDK, earlier there was not much deviation in the server info hence was simpler to add single source for defaults with sync and async servers. 

https://github.com/numaproj/numaflow-rs/pull/71/files

**Comment by kohlisid** [comment]
We could have passed container type as parameter that can give the default with the requested one. 
Will make a PR for any refactor that I can see

**Comment by KeranYang** [comment]
thank you @kohlisid !

**Pull request merged**